ir/json_parser.h

```diff
@@ -56,35 +59,26 @@ class JsonString : public JsonData, public std::string {
     DECLARE_TYPEINFO(JsonString, JsonData);
 };
 
-class JsonVector : public JsonData, public std::vector<JsonData *> {
+class JsonVector : public JsonData, public std::vector<std::unique_ptr<JsonData>> {
  public:
     JsonVector() {}
-    JsonVector(const std::vector<JsonData *> &v)  // NOLINT(runtime/explicit)
-        : std::vector<JsonData *>(v) {}
-    JsonVector &operator=(const JsonVector &) & = default;
-    JsonVector &operator=(JsonVector &&) & = default;
+    JsonVector(std::vector<std::unique_ptr<JsonData>> &&v)  // NOLINT(runtime/explicit)
+        : std::vector<std::unique_ptr<JsonData>>(std::move(v)) {}
+    JsonVector &operator=(const JsonVector &) = delete;
+    JsonVector &operator=(JsonVector &&) = default;
 
     DECLARE_TYPEINFO(JsonVector, JsonData);
 };
 
-class JsonObject : public JsonData, public ordered_map<std::string, JsonData *> {
-    bool _hasSrcInfo = true;
+class JsonObject : public JsonData, public ordered_map<std::string, std::unique_ptr<JsonData>> {
+    // bool _hasSrcInfo = true;
```

Comment:
What happened with src info here? Unused?

Reply:
It turns out to be unneeded -- it was just caching whether or not there was a `"Source_Info"` tag in a JsonObject, which was pointless as we never looked it up more than once anyways.  I commented it out to see what looked at it, and then forgot about it -- I'll delete it completely